### PR TITLE
Feat/edit profile

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
feat:
> Components: PageWrapper, Wrapper (for main content), SideBar, Accordeon, separate form for continue google auth, Edit Profile, PhotoChanger (no logic).
> Pages: EditDoctor, now it is possible to edit local doctor profile data, Help.
> Utils: small custom library to work with cookies. Implemented get, set and delete methods. UseAvoidRenderOnPath hook to prevent component render on array of paths.
> Other: Added Redux Persist globally. Added RTK Query in services folder.

Profile and sidebar

screenshot-1682010899011

Help
screenshot-1681910588109

## Issue ticket code (and/or) and link

> [Link to JIRA ticket]( https://smartdc23.atlassian.net/browse/SMAR-51) 51
> [Link to JIRA ticket]( https://smartdc23.atlassian.net/browse/SMAR-53) 53
> [Link to JIRA ticket](https://smartdc23.atlassian.net/browse/SMAR-95) 95

### General

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [ ] covered by tests
- [x] check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/assets/doctorPicture.png
Comment (lyudad) [inline src/assets/doctorPicture.png:1]: I still think its bad example of default photo

File: src/components/Accordion/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import Wrapper from '@components/Wrapper';
+import React from 'react';
+import picture from '@assets/Sign Out.svg';
+import { AccordionBody, AccordionTitle, Arrow } from './styles';
+
+const Accordion = () => {
+  const [toggle, setToggle] = React.useState(false);
```
Comment (lyudad) [inline src/components/Accordion/index.tsx]: types

File: src/components/Accordion/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const [heightEl, setHeightEl] = React.useState<string>();
+
+  const refHeight = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (refHeight.current)
+      setHeightEl(`${refHeight.current.scrollHeight + 20 + 16}px`);
```
Comment (lyudad) [inline src/components/Accordion/index.tsx]: are you sure you need it? with current list of ready libraries when I see such code usually mean that something custom and bad happend

File: src/components/Accordion/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          height: toggle ? `${heightEl}` : '0px',
+          overflow: 'hidden',
+          transition: '0.3s all',
+        }}
+        ref={refHeight}
+      >
+        Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore,
+        suscipit quae maiores sunt ducimus est dolorem perspiciatis earum
+        corporis unde, dicta quibusdam aut placeat dignissimos distinctio vel
+        quo eligendi ipsam.
```
Comment (lyudad) [inline src/components/Accordion/index.tsx]: all text in text file

File: src/components/Auth/AuthGoogleButton/index.tsx [hunk 4]
```diff
@@ -7,17 +7,18 @@ import {
...
   const { t } = useTranslation();
-  const login = useGoogleLogin({});
+  // const login = useGoogleLogin({});
 
-  const handleLogin = () => {
-    login();
-  };
+  // const handleLogin = () => {
+  //   login();
+  // };
```
Comment (lyudad) [inline src/components/Auth/AuthGoogleButton/index.tsx]: check checklist

File: src/components/Auth/LoginForm/index.tsx [hunk 5]
```diff
@@ -61,15 +68,23 @@ function LoginForm() {
...
+    dispatch(loginQuery(data)).then((res) => {
+      if (!res.error) {
+        console.log(res.payload.token);
+        const token = res.payload.token;
+        const doctor = res.payload.userInfo;
+        cookie.delete('googleUserData');
+        cookie.set('accessToken', token, 1);
+        console.log('doctor', doctor);
+        dispatch(doctorActions.getDoctor(doctor));
+        console.log('DoctorData inside func', doctorData);
```
Comment (lyudad) [inline src/components/Auth/LoginForm/index.tsx]: unacceptable, check checlist

File: src/components/Accordion/index.tsx (hunk 1 above)
Comment (DaniilVolobuiev) [reply src/components/Accordion/index.tsx]: fixed

File: src/assets/doctorPicture.png
Comment (DaniilVolobuiev) [reply src/assets/doctorPicture.png:1]: changed photo

File: src/components/Accordion/index.tsx (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/components/Accordion/index.tsx]: I watched 10 min youtube video how to make accordeon :) I dont think that something bad will happened. It looks simple.

File: src/components/Accordion/index.tsx (hunk 3 above)
Comment (DaniilVolobuiev) [reply src/components/Accordion/index.tsx]: fixed

File: src/components/Auth/AuthGoogleButton/index.tsx (hunk 4 above)
Comment (DaniilVolobuiev) [reply src/components/Auth/AuthGoogleButton/index.tsx]: removed commented code

File: src/components/Auth/LoginForm/index.tsx (hunk 5 above)
Comment (DaniilVolobuiev) [reply src/components/Auth/LoginForm/index.tsx]: removed console.log, moved token age to env

File: src/components/Accordion/index.tsx (hunk 2 above)
Comment (lyudad) [reply src/components/Accordion/index.tsx]: why 20 + 16? why not 36?

File: src/assets/doctorPic.png
Comment (lyudad) [inline src/assets/doctorPic.png:1]: why we still have it?

Comment (DaniilVolobuiev) [reply src/assets/doctorPic.png:1]: deleted

File: src/components/Accordion/index.tsx (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/components/Accordion/index.tsx]: 20 is top margin in accordeonBody, 16 is padding. It doesnt counted with scrollHeight property, so i need to apply it manually. I moved them to variables for better experience.

**Pull request merged**